services-core/src/history/events.rs

```diff
@@ -93,9 +93,11 @@ impl EventRegistry {
         // Create temp file to be written completely before rename
         let temp_file = File::create(&temp_path)
             .with_context(|| format!("couldn't create {}", temp_path.display()))?;
-        let buffered_writer = BufWriter::new(temp_file);
-        self.write_to(buffered_writer)?;
-
+        {
```

Comment:
Wouldn't the context have to include the line above to make sure the file is dropped before the rename?

Reply:
The file is moved into the BufWriter so it will be dropped when the buffered writer is, but I like the suggestion as it makes the code less prone to accidental errors during refactoring.

Reply:
Easy peasy! Great suggestion. Thanks.